File: cypress/cypress/pages/loginPO.js
class loginPO{
    elements ={
        LoginPage_Title_text : () => cy.xpath("//h1[text()='Login Page']"),
        LoginPage_Username_textbox : () => cy.xpath("//input[@id='username']"),       
        LoginPage_Password_textbox : () => cy.xpath("//input[@id='password']"),       
        LoginPage_Login_button : () => cy.xpath("//button[text()='Login']"),       
        //Errors
        LoginPage_Error_InvalidUsername : () => cy.xpath("//b[text()='Your username is invalid!']"),  
        LoginPage_Error_InvalidPassword : () => cy.xpath("//b[text()='Your password is invalid!']"),  
    }
    Verify_Expected_Elements_Appear(){
        this.elements.LoginPage_Title_text().should('be.visible')
        this.elements.LoginPage_Username_textbox().should('be.visible')
        this.elements.LoginPage_Password_textbox().should('be.visible')
        this.elements.LoginPage_Login_button().should('be.visible')
    }   
    Login(username, password) {
        this.elements.LoginPage_Username_textbox().type(username);
        this.elements.LoginPage_Password_textbox().type(password);
        this.elements.LoginPage_Login_button().click();
    }
    Verify_Invalid_Username_Error_Appears(){
        this.elements.LoginPage_Error_InvalidUsername().should('be.visible')
    }
    Verify_Invalid_Password_Error_Appears(){
        this.elements.LoginPage_Error_InvalidPassword().should('be.visible')
    }  
}

module.exports = new loginPO();

require('cypress-xpath')

File: capybara/spec/pages/loginPO.rb
# spec/loginPO.rb
class LoginPO
    include Capybara::DSL
    include RSpec::Matchers
  
    def initialize 
      @LoginPage_Title_text = '//h1[text()=\'Login Page\']' 
      @LoginPage_Username_textbox = '//input[@id=\'username\']'       
      @LoginPage_Password_textbox = '//input[@id=\'password\']'       
      @LoginPage_Login_button = '//button[text()=\'Login\']'       
      #Errors
      @LoginPage_Error_InvalidUsername = '//b[text()=\'Your username is invalid!\']'  
      @LoginPage_Error_InvalidPassword = '//b[text()=\'Your password is invalid!\']'  
    end

    def visit_page
        visit('https://practice.expandtesting.com/login')
      end

    def Verify_Expected_Elements_Appear
      expect(find(:xpath, @LoginPage_Title_text ).visible?).to be(true)
      expect(find(:xpath, @LoginPage_Username_textbox ).visible?).to be(true)
      expect(find(:xpath, @LoginPage_Password_textbox ).visible?).to be(true)
      expect(find(:xpath, @LoginPage_Login_button ).visible?).to be(true)
    end

    def Login(username, password)
      find(:xpath, @LoginPage_Username_textbox).set(username)
      find(:xpath, @LoginPage_Password_textbox).set(password)
      find(:xpath, @LoginPage_Login_button).click
    end

    def Verify_Invalid_Username_Error_Appears
      expect(find(:xpath, @LoginPage_Error_InvalidUsername ).visible?).to be(true)
    end

    def Verify_Invalid_Password_Error_Appears
      expect(find(:xpath, @LoginPage_Error_InvalidPassword ).visible?).to be(true)
    end
  end